.github/workflows/CI.yml

```diff
@@ -0,0 +1,277 @@
+      - name: setup Docker
+        uses: docker-practice/actions-setup-docker@0.0.1
+        with:
+          docker_version: 18.09
+          docker_channel: stable
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+          path: src/github.com/goharbor/harbor
+      - name: setup env
+        run: |
+          pwd
+          docker version
+          go env
+          echo "::set-env name=GOPATH::$(go env GOPATH):$RUNNER_WORKSPACE"
+          echo "::add-path::$(go env GOPATH)/bin"
+          echo "::set-env name=TOKEN_PRIVATE_KEY_PATH::${GITHUB_WORKSPACE}/tests/private_key.pem"
+        shell: bash
+      - name: before_install
+        run: |
+          set -x
+          pwd
+          env
+          #sudo apt install -y xvfb
+          #xvfb-run ls
+          curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+          chmod +x docker-compose
+          sudo mv docker-compose /usr/local/bin
+          IP=`hostname -I | awk '{print $1}'`
+          echo '{"insecure-registries" : ["'$IP':5000"]}' | sudo tee /etc/docker/daemon.json
+          echo "::set-env name=IP::$IP"
+          sudo cp ./tests/harbor_ca.crt /usr/local/share/ca-certificates/
+          sudo update-ca-certificates
+          sudo service docker restart
+      - name: script
+        run: |
+          echo IP: $IP
+          bash ./tests/showtime.sh ./tests/travis/distro_installer.sh
+
+  UI_UT:
+    env:
+      UI_UT: true
+    runs-on:
+      #- self-hosted
+      - ubuntu-latest
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.16.2'
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+          path: src/github.com/goharbor/harbor
+      - name: setup env
+        run: |
+          pwd
+          go env
+          echo "::set-env name=GOPATH::$(go env GOPATH):$RUNNER_WORKSPACE"
+          echo "::add-path::$(go env GOPATH)/bin"
+          echo "::set-env name=TOKEN_PRIVATE_KEY_PATH::${GITHUB_WORKSPACE}/tests/private_key.pem"
+        shell: bash
+      - name: before_install
+        run: |
+          set -x
+          pwd
+          env
+          #sudo apt install -y xvfb
+          #xvfb-run ls
+          #curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+          #chmod +x docker-compose
+          #sudo mv docker-compose /usr/local/bin
+          IP=`hostname -I | awk '{print $1}'`
+          #echo '{"insecure-registries" : ["'$IP':5000"]}' | sudo tee /etc/docker/daemon.json
+          echo "::set-env name=IP::$IP"
+          sudo cp ./tests/harbor_ca.crt /usr/local/share/ca-certificates/
+          sudo update-ca-certificates
+          #sudo service docker restart
+      - name: script
+        run: |
+          echo IP: $IP
+          bash ./tests/showtime.sh ./tests/travis/ui_ut_run.sh
```

Comment:
Please use job matrix to simpler the definitions of jobs.

Reply:
I separate each job definition in order to optimize each one later. Matrix didn't introduce much benefits here.